stratum/public/proto/p4_role_config.proto

```diff
@@ -0,0 +1,44 @@
+// Copyright 2022-present Open Networking Foundation
+// Copyright 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO(max)
+// The P4RoleConfig contains ...
+
+syntax = "proto3";
+
+option cc_generic_services = false;
+
+package stratum;
+
+import "p4/v1/p4runtime.proto";
+
+
+// The P4RoleConfig message conists of these fields:
+//  accessible_p4_ids - a list of the
+//  packet_in_filters -
+//  receives_packet_ins -
+message P4RoleConfig {
+  repeated uint32 accessible_p4_ids = 1;
+  repeated p4.v1.PacketMetadata packet_in_filters = 2;
```

Comment:
do we need to specify somewhere the allowed values for `PacketMetadata` ?

Reply:
Are you thinking in terms of checking the `metadata_id` provided for validity against the p4info? Could be tricky sometimes as role config can happen before pipeline push.

Reply:
Initially, I was thinking having a realm of admissible values to make sure the values used here are "certified". BUT it is not really necessary. Moreover, there are many components that might want to use this - not only sd-fabric.

Probably, the most important thing to do in future is the exclusivity of the accesses to the resources (even though we might need to revisit in future but for now we should not have overlapping roles)